### PR TITLE
DPL-1063 - As a developer I need to create a table to store aliquots to track library volumes and enable top-up management

### PR DESCRIPTION
**Pull request by dasunpubudumal**
Closes #

#### Changes proposed in this pull request

- Create `aliquots` table

#### Instructions for Reviewers

_[All PRs] - Confirm PR template filled_  
_[Feature Branches] - Review code_  
_[Production Merges to `main`]_  
 &nbsp; &nbsp; \- _Check story numbers included_  
 &nbsp; &nbsp; \- _Check for debug code_  
 &nbsp; &nbsp; \- _Check version_

### CODE REVIEW

File: Gemfile.lock
Comment (dasunpubudumal) [inline Gemfile.lock:1]: Please note that these are dependency upgrades from branches `depfu/update/rubocop-performance-1.21.0` and `depfu/update/rubocop-rspec-2.28.0`.

File: db/migrate/20240404142248_create_aliquots.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+class CreateAliquots < ActiveRecord::Migration[7.0]
+  def up
+    create_table :aliquot, primary_key: :id, options: 'CHARSET=utf8 COLLATE=utf8_unicode_ci' do |t|
```
Comment (BenTopping) [inline db/migrate/20240404142248_create_aliquots.rb:3]: `id` is the default primary key so we don't have to declare it. However it does make it more verbose so i'm fine to leave it in.

File: db/migrate/20240404142248_create_aliquots.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+      t.string :lims_source, null: false, comment: 'The LIMS system that the aliquot was created in'
+      t.string :lims_uuid, null: false, comment: 'The UUID of the aliquot in the LIMS system'
+      t.string :aliquot_type, null: false, comment: 'The type of the aliquot'
+      t.string :source_type, null: false, comment: 'The type of the source of the aliquot'
+      t.string :source_barcode, null: false, comment: 'The barcode of the source of the aliquot'
+      t.string :sample_name, null: false, comment: 'The name of the sample that the aliquot was created from'
+      t.string :used_by_type, null: false, comment: 'The type of the entity that the aliquot is used by'
+      t.string :used_by_barcode, null: false, comment: 'The barcode of the entity that the aliquot is used by'
+      t.decimal :volume, precision: 10, scale: 2, null: false, comment: 'The volume of the aliquot'
```
Comment (BenTopping) [inline db/migrate/20240404142248_create_aliquots.rb:12]: We may also want to store concentration and timestamps.

File: spec/models/aliquots_spec.rb
Comment (BenTopping) [inline spec/models/aliquots_spec.rb:1]: There are some shared tests in spec/support that might apply e.g. it_behaves_like_a_singular_resource. Some of the other tests also test 'create_or_update_from_json'. Not 100% sure these are applicable here but worth checking.

File: db/migrate/20240404142248_create_aliquots.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+      t.string :id_lims, null: false, comment: 'The LIMS system that the aliquot was created in'
+      t.string :lims_uuid, null: false, comment: 'The UUID of the aliquot in the LIMS system'
+      t.string :aliquot_type, null: false, comment: 'The type of the aliquot'
+      t.string :source_type, null: false, comment: 'The type of the source of the aliquot'
+      t.string :source_barcode, null: false, comment: 'The barcode of the source of the aliquot'
+      t.string :sample_name, null: false, comment: 'The name of the sample that the aliquot was created from'
+      t.string :used_by_type, null: false, comment: 'The type of the entity that the aliquot is used by'
+      t.string :used_by_barcode, null: false, comment: 'The barcode of the entity that the aliquot is used by'
+      t.decimal :volume, precision: 10, scale: 2, null: false, comment: 'The volume of the aliquot'
+      t.decimal :concentration, precision: 10, scale: 2, null: false, comment: 'The concentration of the aliquot'
```
Comment (andrewsparkes) [inline db/migrate/20240404142248_create_aliquots.rb:13]: @BenTopping 
Just checking whether you are storing concentration or molarity here and if the units are always the same?
Concentrations are typically in ng/ul (nanograms per microlitre). Molarity is usually in nM (nanomolar). But not always.
In Sequencescape this is done as 2 fields, a key ('concentration' or 'molaity') and a units ('ng/ul', 'g/ml', 'nM', 'M', etc.)

Comment (andrewsparkes) [reply db/migrate/20240404142248_create_aliquots.rb:13]: If fixed put the expected units in the comment?

Comment (stevieing) [reply db/migrate/20240404142248_create_aliquots.rb]: We are storing concentration. I have spoken to James Watts and this is deemed useful for calculations. Volume will be uL and concentration ng/ul. As there is no intention to reuse for short read currently we can leave as is.

Comment (dasunpubudumal) [reply db/migrate/20240404142248_create_aliquots.rb]: @stevieing I will add them in the column comments (uL in `volume` column's comment and ng/ul in `concentration` column's comment).

File: Gemfile.lock
Comment (dasunpubudumal) [inline Gemfile.lock:1]: These are some dependency upgrades merged from `depfu` updates.

**Review by stevieing — APPROVED** [review]

**Pull request merged**